docs/drivers/azure.md

```diff
@@ -10,51 +10,105 @@ parent="smn_machine_drivers"
+> **NOTE:** This documentation is for the new version of  Azure driver started
+> shipping with v0.7.0 and it is not backwards-compatible with the old Azure
+> driver. If you like to manage your existing Azure machines, please download
+> and use machine versions prior to v0.7.0.
 
-    $ openssl req -x509 -nodes -days 365 -newkey rsa:1024 -keyout mycert.pem -out mycert.pem
-    $ openssl pkcs12 -export -out mycert.pfx -in mycert.pem -name "My Certificate"
-    $ openssl x509 -inform pem -in mycert.pem -outform der -out mycert.cer
+[azure]: http://azure.microsoft.com/
+[trial]: https://azure.microsoft.com/free/
 
-Go to the Azure portal, go to the "Settings" page (you can find the link at the bottom of the
-left sidebar - you need to scroll), then "Management Certificates" and upload `mycert.cer`.
+## Authentication
 
-Grab your subscription ID from the portal, then run `docker-machine create` with these details:
+First time you try to create a machine, Azure driver will ask you to
+authenticate:
 
-    $ docker-machine create -d azure --azure-subscription-id="SUB_ID" --azure-subscription-cert="mycert.pem" A-VERY-UNIQUE-NAME
+    $ docker-machine create --driver azure <machine-name>
+    Running pre-create checks...
+    Microsoft Azure: To sign in, use a web browser to open the page https://aka.ms/devicelogin.
+    Enter the code [...] to authenticate.
 
-The Azure driver uses the `b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-15_10-amd64-server-20151116.1-en-us-30GB`
-image by default. Note, this image is not available in the Chinese regions. In China you should
- specify `b549f4301d0b4295b8e76ceb65df47d4__Ubuntu-15_10-amd64-server-20151116.1-en-us-30GB`.
+After authenticating, the driver will remember your credentials up to two weeks.
 
-You may need to `machine ssh` in to the virtual machine and reboot to ensure that the OS is updated.
+## Options
 
-Options:
+Azure driver only has a single required argument to make things easier. Please
+read the optional flags to configure machine details and placement further.
 
--   `--azure-docker-port`: Port for Docker daemon.
--   `--azure-image`: Azure image name. See [How to: Get the Windows Azure Image Name](https://msdn.microsoft.com/en-us/library/dn135249%28v=nav.70%29.aspx)
--   `--azure-location`: Machine instance location.
--   `--azure-password`: Your Azure password.
--   `--azure-publish-settings-file`: Azure setting file. See [How to: Download and Import Publish Settings and Subscription Information](https://msdn.microsoft.com/en-us/library/dn385850%28v=nav.70%29.aspx)
--   `--azure-size`: Azure disk size.
--   `--azure-ssh-port`: Azure SSH port.
--   `--azure-subscription-id`: **required** Your Azure subscription ID (A GUID like `d255d8d7-5af0-4f5c-8a3e-1545044b861e`).
--   `--azure-subscription-cert`: **required** Your Azure subscription cert.
--   `--azure-username`: Azure login user name.
+Required:
+
+- `--azure-subscription-id`: **(required)** Your Azure Subscription ID.
```

Comment:
Makes sense for interactive users, but we (and a few others like Rancher and DCHQ) are using this as part of our automated workflows to create Docker hosts.

Is there a solution here?